### ISSUE TITLE
.fromObject() throws, does not return invalid object - is this a feature or a bug?

### ISSUE THREAD

**Issue by balagge** [moment/luxon#568]
`Duration.fromObject()` and `DateTime.fromObject()` throws exception if the object contains invalid keys or values. Is this a deliberate choice? Other constructor methods return an invalid DateTime or Duration object in similar cases.

consider:

```javascript
DateTime.fromObject({foobar:2}); // throws exception
DateTime.fromObject({year:"foobar"}); // throws exception
    // however:
DateTime.fromISO("foobar"); // returns invalid DateTime object
DateTime.fromJSDate("foobar"); // returns invalid DateTime object
DateTime.fromRFC2822("foobar"); // returns invalid DateTime object

    // same with Durations
Duration.fromObject({foobar:2}); // throws exception
Duration.fromObject({year:"foobar"}); // throws exception
    // however:
Duration.fromISO("foobar"); // returns invalid DateTime object
```

This seems odd, unless there is some very specific reason for the `fromObject()` methods to behave differently.

**Comment by GillesDebunne** [comment]
Indeed, this might seem inconsistant. An exception is thrown when the input is of invalid type (unknown field, wrong type...), while an invalid object is returned when the type is correct but the *value* is not.

In version 2, this `invalid` state disappears for `DateTime` and `Duration`.
Invalid input will result in an exception, unless a specific option is provided, in which case builders will return `null` instead. What do you think of this new behavior?

**Comment by icambron** [comment]
I agree this feels a little inconsistent. What Luxon is trying to accomplish here is to differentiate *programmer errors* from *data problems*. Of course, those are only statistical distinctions; data can always be programmatic and programming can always be data-driven. But it was my feeling at the time that having the wrong keys in the object was much *more likely* to be a programmer error and having an incorrectly-formatted string was more likely to be bad data coming from outside the programmers control. So I wanted to throw on programmer errors to make sure the programmer knew, and I wanted to make the data issues get handled by the code not requiring `try{}`.

But like Gilles says, we are reworking all of this in 2.0, mostly because the distinction has been more confusing than helpful and also because "invalid" objects have proved painful in general.

I'm going to close this ticket because I don't foresee us fixing this in 1.x and we have a definite plan to fix it in 2.0.
